catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/suggestion/MgrsCoordinateProcessor.java

```diff
@@ -0,0 +1,355 @@
+
+  /**
+   * Given a list of {@link Suggestion}s and the query that yielded them, enhance the list with
+   * additional suggestions based upon the presence of coordinate literals in the query string.
+   *
+   * @param results the list of results to enhance.
+   * @param query the query string to search for coordinate literals.
+   * @return the enhanced list.
+   */
+  public List<Suggestion> enhanceResults(List<Suggestion> results, String query) {
+    LiteralSuggestion literal = getMgrsSuggestion(query);
+    if (literal != null && literal.hasGeo()) {
+      LOGGER.debug("Adding the MGRS suggestion to results");
+      results.add(0, literal);
+    }
+    return results;
+  }
+
+  /**
+   * Given an input string, return the list of properly formatted MGRS coordinate strings that are
+   * ready for submission to usng4j for mathematical evaluation.
+   *
+   * @param query the query string to search for coordinate literals.
+   * @return a list of valid MGRS strings ready for evaluation.
+   */
+  @VisibleForTesting
+  List<String> getMgrsCoordinateStrings(String query) {
+    return coordinateStringsFromQuery(query);
+  }
+
+  /**
+   * Build the MGRS suggestion.
+   *
+   * @param query the query string to search for coordinate literals.
+   * @return a suggestion object with geometry attached to it; a {@link BoundingBox} if only one
+   *     coordinate was detected, or a collection of {@link LatLon} center points otherwise.
+   */
+  @Nullable
+  private LiteralSuggestion getMgrsSuggestion(String query) {
+    List<String> matches = coordinateStringsFromQuery(query);
+    if (matches.isEmpty()) {
+      LOGGER.debug("No valid MGRS strings were found or could be inferred");
+      return null;
+    }
+    StringBuilder nameBuilder = new StringBuilder("MGRS: ");
+    List<UsngCoordinate> uniqueMatches =
+        matches
+            .stream()
+            .distinct()
+            .map(this::parseMgrsString)
+            .filter(Objects::nonNull)
+            .peek(mgrs -> nameBuilder.append("[ ").append(mgrs.toMgrsString()).append(" ] "))
+            .collect(Collectors.toList());
+    if (uniqueMatches.size() == 1) {
+      UsngCoordinate mgrsCoord = uniqueMatches.get(0);
+      BoundingBox bb = translator.toBoundingBox(mgrsCoord);
+      return new LiteralSuggestion(LITERAL, nameBuilder.toString(), latLonFromBoundingBox(bb));
+    }
+    List<LatLon> geo =
+        uniqueMatches
+            .stream()
+            .map(translator::toLatLon)
+            .map(d -> from(d.getLat(), d.getLon()))
+            .collect(Collectors.toList());
+    return new LiteralSuggestion(LITERAL, nameBuilder.toString(), geo);
+  }
+
+  @Nullable
+  private UsngCoordinate parseMgrsString(String mgrs) {
+    try {
+      return translator.parseMgrsString(mgrs);
+    } catch (ParseException e) {
+      LOGGER.debug(format("Detected string [%s] was not valid MGRS", mgrs), e);
```

Comment:
✏️ Use parameterized logging statements instead of calling `format`

Reply:
Note that I want to take advantage of the signature that allows me to pass an exception. The call to `format` is to ensure I call the version of `.debug(...)` that I want.

Reply:
The logging statement in the code above uses brackets around the string; otherwise, they are equivalent.
```suggestion
      LOGGER.debug("Detected string [{}] was not valid MGRS", mgrs, e);
```

Reply:
The signature of the above method call identifies `e` as some object, and not a `Throwable`. 
https://www.slf4j.org/apidocs/org/slf4j/Logger.html#debug(java.lang.String,%20java.lang.Object,%20java.lang.Object)

This is the version of the method that I want
https://www.slf4j.org/apidocs/org/slf4j/Logger.html#debug(java.lang.String,%20java.lang.Throwable)

Reply:
https://www.slf4j.org/faq.html#paramException

Reply:
Since `LOGGER.debug("Detected string {} was not valid MGRS", mgrs, e);` will print the stack trace of `e`, I see not reason to use `format` here.  I see a reason not to use it, since it is extra processing for no gain.

Reply:
Nice. Will fix 👍